terraform/eval_context.go

```diff
@@ -44,11 +44,11 @@ type EvalContext interface {
 	// that owns the given provider before calling this method.
 	Provider(addrs.AbsProviderConfig) providers.Interface
 
-	// ProviderSchema retrieves the schema for a particular provider, which
-	// must have already been initialized with InitProvider.
+	// ProviderSchema retrieves the schema for a particular provider, which must
+	// have already been initialized with InitProvider.
 	//
-	// This method expects an _absolute_ provider configuration address, since
-	// resources in one module are able to use providers from other modules.
+	// This method expects a provider FQN string, since resources in one module
```

Comment:
This taking a provider FQN rather than a provider configuration address makes sense to me, because it's impossible for the schema of a provider to vary based on how it's configured (schema happens before configuration). With that said, could we change the argument type to `addrs.Provider` here so we can have the Go compiler check that we updated them all correctly, and that we continue to get this right moving forward?